snappy_wrappers/wrappers/somatic_variants_qc/summarize-vcf.py

```diff
@@ -0,0 +1,269 @@
+# import packages
+
+import json  # for writing out the json file
+from cyvcf2 import VCF  # for reading vcf file
+import tabix
+from optparse import OptionParser
+
+
+parser = OptionParser()
+parser.add_option("--rawvcf", help="raw vcf file")
+parser.add_option("--filtered-vcf", help="filtered vcf file")
+parser.add_option(
+    "--exom-bedfile",
+    help="exom regions bed file",
+)
+parser.add_option("--padding", help="it's padding", default=0)
+parser.add_option(
+    "--repeated-region-bedfile",
+    help="bed file contains repeated or difficult to map regions",
+)
+parser.add_option(
+    "--output",
+    help="json file contains information about vcf file",
+)
+(options, args) = parser.parse_args()
+
+# from pybedtools import Interval,BedTool
+
+# I will use for loop here
+# It could be better by using multi threading.
+# TODO:
+# - Number of variants called (from the `*.full.vcf.gz`file) [X] Get all variants from full file
+# - Number of variants that pass all filters [X] Get all variants from fitlered file
+# - Number of variants passing the filters that are inside & outside of the exome bed file (with padding) [X]
+# - Number of variants in repeated or other difficult to map regions [ ] Need a bed file containing repeated
+# - Number of SNVs & indels [X]
+# - Indel lengths statistics [X]
+# - Number of variants with more than 1 ALT allele (I _think_ that this value should be 0 for all variants passing the filters) [ ]
+# - Number of SNVs with minimal support (only one read supporting the variant), possibly separating inside & outside exome bed [X]
+# - Number of SNVs with limited support (5 reads or less supporting the variant), possibly separating inside & outside exome bed [X]
+# - Number of SNVs with strong support (at least 10 reads, VAF greater or equal to 10%) [X] VAF need to be joined
+# - Number of SNVs in each mutation class (C>A, C>G, C>T, T>A, T>C, T>G), for minimal, limited, average & strong support [X] not for minimal,limited,average yet
+# - Metrics on strand bias could also be collected (F1R2 & F2R1 vcf FORMAT) [ ] PAUSE
+# - the VAF distribution. Perhaps we can split the variants (both SNVs & indels) into those with very little support (<1%), subclonal (<10%), the main part (<40%), affected by CNV (>40%)[ ]
+
+
+# Adding A>C G>T grouping
+# Table for VAF
+# Genotype check > report different genotypes
+# Checking number multi allelic
+# Support alternative allel in the normal
+# #alt in normal + average ALT + max ALT
+# snappy pipeline for this
+
+##### SNAPPY PIPELINE
+## WRAPPER
+# - Checking the configuration file
+# - Checking input file
+## PIPELINE
+# - __init__ :
+
+CONTIG = ["chr" + str(i) for i in range(1, 22)]
```

Comment:
You don't want to encode the contig names here. For GRCh37, the chromosomes are named `1` to `22`, `X` & `Y`, without the `chr` prefix. Also, what happens with the mitochondrial sequence? And what if you want to analyse mouse data?